charmpath.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrepo
+
+import (
+	"os"
+	"path/filepath"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+)
+
+// CharmPath is used to access a charm at a given path.
+type CharmPath interface {
+	// Charm returns the charm represented by this path.
+	// If the series is not specified, the charm's default
+	// series is used, if any. Otherwise, the series is
+	// validated against those the charm declares it supports.
+	Charm(series string) (charm.Charm, *charm.URL, error)
+}
+
+// charmPath represents a path (eg local directory) containing a
+// single charm.
+type charmPath struct {
+	Name      string
+	CharmInfo charm.Charm
+}
+
+var _ CharmPath = (*charmPath)(nil)
+
+// NewCharmPath creates and returns an instance used to
+// open a charm located at the given path.
+func NewCharmPath(path string) (CharmPath, error) {
```

Comment:
I see where you're coming from but don't particularly like:
url.Series, err = charm.SeriesForCharm(requestedSeries, c.Meta().Series)

as to breaks encapsulation somewhat.

Ok, so what I've done to try and get this accepted is add series to NewCharmAtPath

func NewCharmAtPath(path, series string) (charm.Charm, *charm.URL, error)

It doesn't allow a charm for a series to be peeled off a previously parsed path, but it fits with how core calls this functionality which is simply to get a charm for a given series and then use it.